hack/lrp-test.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# Leader retry period (LRP) test
+# Iterate over LRP from 15ms to 500ms.
+# For each LRP value create a number of pods based on the number of nodes (N) and a capcity limit per node (C).
+
+set -o errexit
+set -o pipefail
+set -o xtrace
+
+C=100 # capacity per node
+N=30  # number of compute nodes
+
+## please update following setting to suit the interface name on your host.
+HOST_DEVICE_NAME_FOR_BRIDGE=""
+[ -z "$HOST_DEVICE_NAME_FOR_BRIDGE" ] && echo "define HOST_DEVICE_NAME_FOR_BRIDGE in the script" && exit 1
+
+FILE_PATH=$(dirname "$(readlink --canonicalize "${BASH_SOURCE[0]}")")
+ROOT=$(readlink --canonicalize "$FILE_PATH/..")
+TEMPLATING_BIN_PATH="$ROOT/build/bin/templating-device"
+TEMPLATES_INPUT_DIR_PATH="$ROOT/templates"
+TEMPLATES_OUTPUT_DIR_PATH="$ROOT/deployment"
+
+# leader retry period in miliseconds - included default value of 500 to give baseline
+leader_retry_period=(15 60 120 260 500)
+# total number of pods to be created in a single replica set
+pods_creation_total=($(echo 0.2*$C*$N/1 | bc) $(echo 0.4*$C*$N/1 | bc) $(echo 0.8*$C*$N/1 | bc)) # $(echo 1*$C*$N/1 | bc))
+
+# initial condition checks
+[[ ! -f "$TEMPLATING_BIN_PATH" ]] && echo "have you built the project? Templating binary not found" && exit 2
+[[ ! $(type -P kubectl) ]] && echo "you need kubectl in PATH" && exit 3
+
+rm -rf --preserve-root "$TEMPLATES_OUTPUT_DIR_PATH"
+mkdir -p "$TEMPLATES_OUTPUT_DIR_PATH"
+
+for lrp in ${leader_retry_period[@]}; do
+  for no_pods in ${pods_creation_total[@]}; do
+    echo -e "Test time $(date):\nLeader retry period: $lrp\nNumber of pods in replica: $no_pods"
+    # Generate test artifacts
+    "$ROOT"/build/bin/templating-device --ipam-range 10.10.0.0/16 --number-of-replicas "$no_pods" --input-dir "$TEMPLATES_INPUT_DIR_PATH" --output-dir "$TEMPLATES_OUTPUT_DIR_PATH" --retry-period "$lrp" --lower-device "$HOST_DEVICE_NAME_FOR_BRIDGE"
```

Comment:
This line is important, imo. 

Would you turn it into multi-line so readers can see the full thing without scrolling ?